aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -2604,8 +2607,10 @@ private int processPendingSessions(
                 {
                     case CLIENT:
                     {
+                        nextCommittedSessionId = session.id() + 1;
```

Comment:
This line needs to be moved inside of the if statement that follows it, i.e. after a session was appended to the log.
The `nextCommittedSessionId` needs to be protected against going backwards, i.e. pending sessions are processed from newest to oldest so the subsequent session will decrease the value.
Instead we should also check that the value is larger, i.e. add `if (session.id() >= nextCommittedSessionId)`.